modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/GwsEpics.scala

```diff
@@ -0,0 +1,58 @@
+package edu.gemini.seqexec.server
+
+import java.util.logging.Logger
+
+import squants.{Temperature, Velocity}
+import edu.gemini.epics.acm.{CaService, XMLBuilder}
+import squants.motion.{Bars, MetersPerSecond, Pressure}
+import squants.MetricSystem.Milli
+import squants.space.{Angle, Degrees}
+import squants.thermal.Celsius
+
+/**
+  * GwsEpics wraps the non-functional parts of the EPICS ACM library to interact with the Weather Server.
+  * It has all the objects used to read TCS status values and execute TCS commands.
+  *
+  * Created by jluhrs on 3/10/17.
+  */
+final class GwsEpics private (epicsService: CaService) {
+  private val state = epicsService.getStatusAcceptor("gws::state")
+
+  def humidity: Option[Double] = Option(state.getDoubleAttribute("humidity").value).map(_.doubleValue)
```

Comment:
The `getXXXAtribute` methods return `null` if there is a problem. So, if there is a problem, the end result is `None`.